### PR TITLE
Use `insta::glob` instead of `fixture` macro

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This PR replaces our custom `fixtures` macro used by the formatter with insta's `glob` macro. 

The main motivation for the change is that the `#[fixture]` macro was somewhat error prone because it relied to come before the `#[test]` macro. 

The downside of using `insta::glob` is that it now defines a single test. Maning, all fixture tests run sequentially, rather than in parallel. The slowdown isn't significant on my machine. 

I further improved the snapshot tests by 

* Making the fixture tests integration tests to avoid snapshot name collisions (and these tests are integration tests after all).
* Removing the module name
* Hide the expression (which is always `expression`)

## Test Plan

No functional change.

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5364** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5364" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/5364?utm_source=stack-comment).

File: crates/ruff_python_formatter/tests/fixtures.rs [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+use similar::TextDiff;
+use std::fmt::{Formatter, Write};
+use std::fs;
+
+#[test]
+fn black_compatibility() {
+    insta::glob!(
+        "../resources",
+        "test/fixtures/black/**/*.py",
+        |input_path| {
```
Comment (konstin) [inline crates/ruff_python_formatter/tests/fixtures.rs:11]: Could you move these into their own function? In my experience macro calls often break tooling inside of it

**Review by konstin — APPROVED** [review]

File: crates/ruff_python_formatter/tests/fixtures.rs (hunk 1 above)
Comment (MichaReiser) [reply crates/ruff_python_formatter/tests/fixtures.rs]: It worked 😮

**Comment by davidszotten** [comment]
need to be careful when rebasing past this i guess

File: crates/ruff_python_formatter/tests/fixtures.rs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            write!(snapshot, "{}", Header::new("Ruff Output")).unwrap();
+            write!(snapshot, "{}", CodeFrame::new("py", formatted_code)).unwrap();
+
+            write!(snapshot, "{}", Header::new("Black Output")).unwrap();
+            write!(snapshot, "{}", CodeFrame::new("py", &expected_output)).unwrap();
+
+            insta::with_settings!({
+                omit_expression => true,
+                input_file => input_path,
+                prepend_module_to_snapshot => false,
```
Comment (konstin) [inline crates/ruff_python_formatter/tests/fixtures.rs:76]: thanks this make it so much more readable

**Pull request merged**